plugins/vpp/wireguardplugin/wireguardplugin.go

```diff
@@ -0,0 +1,72 @@
+//  Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at:
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+//go:generate descriptor-adapter --descriptor-name Peer --value-type *vpp_wg.Peer --meta-type *wgidx.WgMetadata --import "go.ligato.io/vpp-agent/v3/plugins/vpp/wireguardplugin/wgidx" --import "go.ligato.io/vpp-agent/v3/proto/ligato/vpp/wireguard" --output-dir "descriptor"
+
+package wireguardplugin
+
+import (
+	"github.com/pkg/errors"
+	"go.ligato.io/cn-infra/v2/health/statuscheck"
+	"go.ligato.io/cn-infra/v2/infra"
+	"go.ligato.io/vpp-agent/v3/plugins/govppmux"
+	kvs "go.ligato.io/vpp-agent/v3/plugins/kvscheduler/api"
+	"go.ligato.io/vpp-agent/v3/plugins/vpp/ifplugin"
+	"go.ligato.io/vpp-agent/v3/plugins/vpp/wireguardplugin/descriptor"
+	"go.ligato.io/vpp-agent/v3/plugins/vpp/wireguardplugin/descriptor/adapter"
+	"go.ligato.io/vpp-agent/v3/plugins/vpp/wireguardplugin/vppcalls"
+
+	_ "go.ligato.io/vpp-agent/v3/plugins/vpp/wireguardplugin/vppcalls/vpp2009"
+)
+
+type WgPlugin struct {
+	Deps
+	// handler
+	WgHandler vppcalls.WgVppAPI
+
+	peerDescriptor *descriptor.WgPeerDescriptor
+}
+
+type Deps struct {
+	infra.PluginDeps
+	KVScheduler kvs.KVScheduler
+	VPP         govppmux.API
+	IfPlugin    ifplugin.API
+	StatusCheck statuscheck.PluginStatusWriter // optional
+}
+
+func (p *WgPlugin) Init() (err error) {
+	// init Wg handler
+	p.WgHandler = vppcalls.CompatibleWgVppHandler(p.VPP, p.IfPlugin.GetInterfaceIndex(), p.Log)
+	if p.WgHandler == nil {
+		return errors.New("Wireguard handler is not available")
+	}
```

Comment:
Agent with VPP 20.05 does not want to start.
```
time="2020-10-05 15:05:44.23681" level=debug msg="-> Init(): vpp-wg-plugin" loc="agent/agent.go(220)" logger=agent
time="2020-10-05 15:05:44.23689" level=debug msg="VPP handler wg incompatible with 20.09-rc0 (12 messages)" loc="logging/log_api.go(37)" logger=defaultLogger
time="2020-10-05 15:05:44.23693" level=debug msg="no compatible version found for handler wg: incompatible handler" loc="logging/log_api.go(37)" logger=defaultLogger
time="2020-10-05 15:05:44.23703" level=fatal msg="Wireguard handler is not available" loc="vpp-agent/main.go(87)" logger=defaultLogger
time="2020-10-05 15:05:45.73858" level=info msg="Signal terminated received, stopping." loc="agent/agent.go(196)" logger=agent
```
Maybe log and return no error here?

Reply:
@ondrej-fabry

Reply:
Add check at the start of `Init()`:
```go
if !p.VPP.IsPluginLoaded("wireguard") {
  p.Log.Warnf("VPP plugin wireguard was disabled by VPP")
  return nil
}
```

similar to NAT plugin.

Reply:
@glazychev-art could you please fix this along with my [previous comment](https://github.com/ligato/vpp-agent/pull/1731#issuecomment-703579972)?

Reply:
@ondrej-fabry Sorry for waiting. I fixed your comments.